jwst/assign_wcs/assign_wcs_step.py

```diff
@@ -65,7 +66,8 @@ def process(self, input):
             # Get the MSA metadata file if needed and add to reffiles
             msa_metadata_file = input_model.meta.instrument.msa_metadata_file
             if msa_metadata_file is not None:
-                msa_metadata_file = self.make_input_path(msa_metadata_file)
+                input_dir = os.path.split(self.make_input_path(input))[0]
+                msa_metadata_file = os.path.join(input_dir, msa_metadata_file)
```

Comment:
Shouldn't this change be unnecessary if the code in `Step` is modified, as you've done below?  I believe the purpose of `Step.input_dir` was to avoid having to do this `os.path` stuff within each step's `process` method.

Further, I suspect we should be calling `process` directly.  Instead, we should be using `run` or `call`, as both of those eventually call `process` after doing the necessary setup of the step, which includes setting `Step.input_dir`.